### PR TITLE
Run codespell in CI

### LINKED ISSUE #143: CI: use codespell

### PR DESCRIPTION
**Pull request by axic**
Closes #143.

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -12,6 +12,17 @@ jobs:
           find examples include lib test -name '*.hpp' -o -name '*.cpp' -o -name '*.h' -o -name '*.c' | xargs clang-format -i
           git diff --color --exit-code
 
+  codespell:
+    docker:
+    - image: circleci/python
+    steps:
+    - checkout
+    - run:
+        name: "Run codespell"
```
Comment (chfast) [inline circle.yml]: Include it in the `lint` job.

File: include/evmc/evmc.h [hunk 2]
```diff
@@ -324,7 +324,7 @@ struct evmc_result
     /**
      * The reference to output data.
      *
-     *  The output contains data coming from RETURN opcode (iff evmc_result::code
+     *  The output contains data coming from RETURN opcode (if evmc_result::code
```
Comment (chfast) [inline include/evmc/evmc.h]: This is not a typo, it means "if and only if".

Comment (axic) [reply include/evmc/evmc.h]: I wasn't sure if you actually `iff` or just a typo. I can add it to a whitelist.

**Comment by axic** [comment]
With pip3:

```
Collecting codespell
  Downloading https://files.pythonhosted.org/packages/fa/75/b28d76bf412bc8e6253b10b7d148486d9e3395b3806a7301251eaaaa835e/codespell-1.13.0.tar.gz (87kB)
    100% |################################| 92kB 5.9MB/s 
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
    ImportError: No module named 'setuptools'
```

Adding `sudo apt-get install python3-setuptools`

**Comment by axic** [comment]
Are you sure it is worth the effort merging the two CI jobs?

```
Package python3-setuptools is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'python3-setuptools' has no installation candidate
```

**Comment by chfast** [comment]
The magic commands are:
```
sudo pip3 install --upgrade pip setuptools
sudo pip3 install codespell
```

**Pull request merged**